src/test/java/seedu/address/logic/commands/SearchCommandTest.java

```diff
@@ -0,0 +1,105 @@
+package seedu.address.logic.commands;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.TypicalBooks.getTypicalBookShelf;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import seedu.address.logic.CommandHistory;
+import seedu.address.logic.UndoRedoStack;
+import seedu.address.logic.commands.SearchCommand.SearchDescriptor;
+import seedu.address.model.BookShelf;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.testutil.SearchDescriptorBuilder;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for SearchCommand.
+ */
+public class SearchCommandTest {
+
+    private Model model;
+
+    @Before
+    public void setUp() {
+        model = new ModelManager(getTypicalBookShelf(), new UserPrefs());
+    }
+
+    @Test
+    public void execute_allFieldsSpecifiedWithSearchTerm_success() {
+        SearchDescriptor searchDescriptor = new SearchDescriptorBuilder().withTitle("1")
+                .withCategory("1").withIsbn("1").withAuthor("1").withSearchTerm("searchterm").build();
+        SearchCommand command = prepareCommand(searchDescriptor);
+
+        Model expectedModel = new ModelManager(new BookShelf(model.getBookShelf()), new UserPrefs());
+
+        assertCommandSuccess(command, model, SearchCommand.MESSAGE_SEARCHING, expectedModel);
+    }
+
+    @Test
+    public void execute_allFieldsSpecifiedNoSearchTerm_success() {
+        SearchDescriptor searchDescriptor = new SearchDescriptorBuilder().withTitle("1")
+                .withCategory("1").withIsbn("1").withAuthor("1").build();
+        SearchCommand command = prepareCommand(searchDescriptor);
+
+        Model expectedModel = new ModelManager(new BookShelf(model.getBookShelf()), new UserPrefs());
+
+        assertCommandSuccess(command, model, SearchCommand.MESSAGE_SEARCHING, expectedModel);
+    }
+
+    @Test
+    public void execute_someFieldsSpecifiedNoSearchTerm_success() {
+        SearchDescriptor searchDescriptor = new SearchDescriptorBuilder().withTitle("1").withIsbn("1").build();
+        SearchCommand command = prepareCommand(searchDescriptor);
+
+        Model expectedModel = new ModelManager(new BookShelf(model.getBookShelf()), new UserPrefs());
+
+        assertCommandSuccess(command, model, SearchCommand.MESSAGE_SEARCHING, expectedModel);
+    }
+
+    @Test
+    public void execute_noFieldSpecifiedNoSearchTerm_success() {
+        SearchDescriptor searchDescriptor = new SearchDescriptorBuilder().build();
+        SearchCommand command = prepareCommand(searchDescriptor);
```

Comment:
Expect NullPointerException?

Reply:
No it just returns an empty `SearchDescriptor`.